Documentation/proposals/202405-scrapeconfig-graduation.md

```diff
@@ -0,0 +1,188 @@
+- Maintain naming conventions and field consistency across existing and new Service Discoveries.
+- Identify inconsistencies in validations.
+
+## Audience
+
+- Prometheus Operator maintainers and contributors.
+- Users and developers relying on the Prometheus Operator for monitoring configurations.
+- Stakeholders interested in the evolution and improvement of the Prometheus Operator.
+
+## Non-Goals
+
+- Implementing all 28 Service Discovery configurations immediately.
+- To plan out a detailed graduation strategy.
+
+## How
+
+In order to provide a more comprehensive and versatile monitoring solution, enhancing the
+Service Discovery support in the Prometheus Operator is crucial. This will address current
+limitations and better serve diverse user needs. The following steps outline our approach to achieving this goal:
+
+#### Additional Service Discovery Configurations
+
+By adding more Service Discovery configurations, we increase the flexibility and utility of the CRD for various user scenarios.
+We believe that most of the user's needs can be solved with the Kubernetes service discovery and the other existing service discoveries that we offer
+and so implementing all the Service discoveries is not a priority. However, this does not imply service discovery support is halted at this number.
+It would be a better use of time to add the service discoveries that the users need as we get feature requests for them. As a starting point,
+we shall add the following 3 Service discoveries, the choice for which is not necessarily backed up by any usage data.
+
+The following is the list of Service Discoveries which we want to support before graduation:
+
+- *`azure_sd_config`*
+- *`consul_sd_config`*
+- *`digitalocean_sd_config`*
+- *`docker_sd_config`*
+- *`dockerswarm_sd_config`*
+- *`dns_sd_config`*
+- *`ec2_sd_config`*
+- *`openstack_sd_config`*
+- *`puppetdb_sd_config`*
+- *`file_sd_config`*
+- *`gce_sd_config`*
+- *`hetzner_sd_config`*
+- *`http_sd_config`*
+- *`kubernetes_sd_config`*
+- *`kuma_sd_config`*
+- *`lightsail_sd_config`*
+- *`linode_sd_config`*
+- *`nomad_sd_config`*
+- *`eureka_sd_config`*
+
+The following is a list of Service Discoveries we will not be supporting yet:
+*Note: this list is tentative in the sense that Service Discoveries in this list might be added based on feature requests but we do no yet
+readily plan to add them.*
+
+- **`ovhcloud_sd_config`**
+- **`ionos_sd_config`**
+- **`scaleway_sd_config`**
+- **`uyuni_sd_config`**
+- **`vultr_sd_config`**
+
+The following Service Discoveries most likely will not be supported due to them being deprecated or inactive:
+
+- **`marathon_sd_config`**
+- **`nerve_sd_config`**
+- **`serverset_sd_config`**
+- **`triton_sd_config`**
+
+#### Expose Missing Fields
+
+- We will incorporate all fields present in Prometheus `scrape_config` into the ScrapeConfig CRD. However, there are specific exceptions:
```

Comment:
Add unified definition top-level configuration.
ref: https://github.com/prometheus-operator/prometheus-operator/issues/6486

Reply:
I understand the issue you have referenced. But could you elaborate a bit on what you are implying? just to be clear we are on the same page.